third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1743,15 +1831,16 @@ fn check_depth_of_type_impl(
 
 /// Represents the execution context for a function. When calls are made, frames are
 /// pushed and then popped to/from the call stack.
-struct Frame {
+pub(crate) struct Frame {
     pc: u16,
+    pub(crate) ty_builder: TypeBuilder,
     // Currently being executed function.
-    function: Rc<LoadedFunction>,
+    pub(crate) function: Rc<LoadedFunction>,
```

Comment:
we might be losing some bottlenecking here making things pub(crate). Might want to hide some of these behind trivial APIs, except things that we are clearly okay with directly accessing everywhere (i.e. if anything needs to maintain consistency or any other type of invariants / guarantees at all, usually better not to allow less direct accesses - obviously can still do within a file here, and a pretty huge file, but we could also move it to a new file if this logic actually applied).

Reply:
Indeed! Removed some of crate visibilities, but not all. The way the code is set up, you cannot hide &mut pub crate access behind a function because then the borrow checker complains, e.g.
```
let code = frame.function().code(); // immutable borrow of frame
...
frame.increment_pc(); // mutable borrow of frame
```

We probably need to refactor other parts first 🙃 so I left locals, pc, frame cache and function with crate visibility for now